course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt

```diff
@@ -249,86 +253,114 @@ class CourseContainerViewModel(
+                state.copy(isCalendarSyncEnabled = isCalendarSyncEnabled())
+            }
+            if (resumeBlockId.isNotEmpty()) {
+                viewModelScope.launch {
+                    delay(500L)
+                    courseNotifier.send(CourseOpenBlock(resumeBlockId))
+                }
+            }
+            if (isIAPFlow) {
+                viewModelScope.launch {
+                    if (isExpiredCoursePurchase) {
+                        if (eventLogger.isSilentIAPFlow.isNull()) {
+                            eventLogger.upgradeSuccessEvent()
                         }
-                        if (isIAPFlow) {
-                            if (isExpiredCoursePurchase) {
-                                if (eventLogger.isSilentIAPFlow.isNull()) {
-                                    eventLogger.upgradeSuccessEvent()
-                                }
-                                _uiMessage.emit(
-                                    UIMessage.ToastMessage(
-                                        resourceManager.getString(
-                                            CoreR.string.iap_success_message
-                                        )
-                                    )
+                        _uiMessage.emit(
+                            UIMessage.ToastMessage(
+                                resourceManager.getString(
+                                    CoreR.string.iap_success_message
                                 )
-                            } else {
-                                iapNotifier.send(CourseDataUpdated())
-                            }
-                            _iapState.value = IAPUIState.CourseDataUpdated
-                        }
-                        _dataReady.value = true
+                            )
+                        )
+                    } else {
+                        iapNotifier.send(CourseDataUpdated())
                     }
-                } ?: run {
-                    _courseAccessStatus.value = CourseAccessError.UNKNOWN
                 }
-            } catch (e: Exception) {
-                logger.e(
-                    throwable = e, metadata = mapOf(
-                        "courseId" to courseId,
-                        "isIAPFlow" to isIAPFlow,
-                        "isExpiredCoursePurchase" to isExpiredCoursePurchase
-                    )
-                )
-                _courseAccessStatus.value = CourseAccessError.UNKNOWN
-                _showProgress.value = false
+                _iapState.value = IAPUIState.CourseDataUpdated
             }
+            _dataReady.value = true
         }
     }
 
+    private fun handleFetchError(e: Throwable) {
+        logger.e(throwable = e)
+        if (!isNetworkRelatedError(e)) {
+            _courseAccessStatus.value = CourseAccessError.UNKNOWN
+        }
+        _showProgress.value = false
+    }
+
+    private fun isNetworkRelatedError(e: Throwable): Boolean {
```

Comment:
The method name does not reflect its functionality, since it doesn't check the network status.